controllers/grocerieController.js

```diff
@@ -60,20 +59,62 @@ const addItemToList = async (req, res) => {
   }
 };
 
+const checkedNotOnList = async (req, res) => {
+  try{
+    const groceries = await SavedList.find({checked: true}).populate('item');
+    console.log(groceries)
+    let deleted;
+    groceries.forEach( async item => {
+    deleted = await Grocerie.updateOne({name:item.item.name}, {$set:{onList:false}});  
+  })
+res.json(deleted);
+} catch (err) {
+  console.log(`checkedNotOnList ERROR ${err}`);
+}
+}
+
+const allNotOnList = async (req, res) => {
+  try{
+    const groceries = await SavedList.find({}).populate('item');
+const deleted = await Grocerie.updateMany({groceries}, {onList:false});
+res.json(deleted);
+} catch (err) {
+  console.log(`checkedNotOnList ERROR ${err}`);
+} 
+}
+
+const itemAddedToList = async (req, res) => {
+  const id = req.params.id;
+  let result;
+  try{
+  const item = await Grocerie.findById(id);
+  if(item.onList) {
+  result = await Grocerie.updateOne({_id: id}, {$set: {onList: false}})
+}
+if(!item.onList){
+  result =   await Grocerie.updateOne({_id: id}, {$set: {onList: true}})
+}
+res.json(result)
+}
+catch (err) {
+  console.log(`itemAddedToList ERROR ${err}`);
```

Comment:
don't you have morgan logger for those?

Reply:
and why not prompting the result error (e.g. `res.status(500).json({error:"bla"})`) . this should be part of the API. and console won't help anyone but the developer.